### PR TITLE
Distributor limitations with Outbox and v6 considerations

### PR DESCRIPTION
**Pull request by ramonsmits**

### CODE REVIEW

**Comment by SzymonPobiega** [comment]
Is this still WIP? Do you need a review?

File: samples/scaleout/distributor-upgrade/sample.md [hunk 1]
```diff
@@ -21,6 +23,17 @@ This sample uses the same solution as the Version 5 [distributor sample](/sample
 
 ## Upgrade
 
+NServiceBus 6 does not have a distributor role. The distributor remains on NServiceBus 5.x, the workers can upgrade to 6.x.
+
+
+Warning: It is not recommended to use Outbox on the workers. Enabling Outbox can result in increasing the enlisted capacity of a worker when errors occur. The distributor model is originally designed to work with distributed transactions (MSDTC). Consider using [Sender Side Distribution](/transports/msmq/sender-side-distribution) instead. When enabling Outbox on a worker, explicitly set transaction mode to "Sends atomic with receive" as the Outbox sets the default transaction mode to "Receive Only".
```
Comment (SzymonPobiega) [inline samples/scaleout/distributor-upgrade/sample.md]: I think this section should be in the general distributor doco, not in the upgrade guide. If somebody has a working V5 solution with a distributor, he has these settings set correctly.

File: samples/scaleout/distributor-upgrade/sample.md [hunk 2]
```diff
@@ -21,6 +23,17 @@ This sample uses the same solution as the Version 5 [distributor sample](/sample
...
+```
+var transport = endpointConfiguration.UseTransport<MsmqTransport>();
+transport.Transactions(TransportTransactionMode.None);
```
Comment (SzymonPobiega) [inline samples/scaleout/distributor-upgrade/sample.md]: None -> SendsAtomicWithReceive

File: samples/scaleout/distributor-upgrade/sample.md [hunk 3]
```diff
@@ -45,17 +58,18 @@ In the original sample the sender was sending messages to the master (Worker1).
 snippet: SenderRouting
 
 
-### Upgrading the workers
+### Upgrading a worker
+
+The `NServiceBus` package needs to be upgraded to 6.x. Workers do no longer need the `NServiceBus.Distributor.MSMQ` package as the worker logic is embedded in the NServiceBus v6 package.
```
Comment (SzymonPobiega) [inline samples/scaleout/distributor-upgrade/sample.md]: `do no longer need` -> `no longer need`
`NServiceBus v6 package` -> `Version 7`

File: samples/scaleout/distributor-upgrade/sample.md [hunk 4]
```diff
@@ -45,17 +58,18 @@ In the original sample the sender was sending messages to the master (Worker1).
...
 
-As both the workers are now only processing messages, they no longer need the `NServiceBus.Distributor.MSMQ` package. The `NServiceBus` package needs to be upgraded to 6.x.
+Note: Endpoints need to be upgraded to version 6 like any other endpoint. Please read our [Upgrade Guides](/nservicebus/upgrades/) for additional information.
```
Comment (SzymonPobiega) [inline samples/scaleout/distributor-upgrade/sample.md]: `version` -> `Version`

**Comment by ramonsmits** [comment]
@SzymonPobiega Please review again, I've addressed your review feedback.

File: samples/scaleout/distributor-upgrade/sample.md [hunk 5]
```diff
@@ -10,6 +10,8 @@ related:
 - samples/scaleout/distributor
 ---
 
+Note: This solution should be seen as a temporary solution. When all endpoints have been upgraded to version 6 the distributor should be migrated to [Sender Side Distribution](/transports/msmq/sender-side-distribution).
```
Comment (SzymonPobiega) [inline samples/scaleout/distributor-upgrade/sample.md]: Is this the new guidance? It contradicts what is written here https://docs.particular.net/transports/msmq/scaling-out

**Comment by ramonsmits** [comment]
Added row to comparison of SSD vs Distributor regarding version compatibility. I think info on SSD and distributor should be present on all versions of the scale-out page. I don't think this page should be version specific.

File: transports/msmq/scaling-out_overview_core_6.partial.md [hunk 6]
```diff
@@ -14,8 +14,9 @@ Following table compares both approaches:
 | No [flow control](https://en.wikipedia.org/wiki/Flow_control_%28data%29)| [Flow control](https://en.wikipedia.org/wiki/Flow_control_%28data%29) via ready messages             |
 | No automatic worker failure detection       | Worker failure detection via ready messages |
 | Round-robin load balancing                  | Fair load balancing                         |
+| Available since core v6 & msmq v1           | Unsupported since core v7 / msmq v1         |
```
Comment (SzymonPobiega) [inline transports/msmq/scaling-out_overview_core_6.partial.md]: There should be a separate partial for V7 that says that SSD is the only option supported. For V7 users there is no need to read this table only to learn that the SSD is their only option.

**Comment by ramonsmits** [comment]
@SzymonPobiega can you re-review?

**Comment by SzymonPobiega** [comment]
@ramonsmits the content is fine but there are still some link issues.

**Pull request merged**